src/DssSpell.t.sol

```diff
@@ -609,541 +534,19 @@ contract DssSpellTest is DssSpellTestBase {
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
-
-        assertEq(rwaurn_008.can(RWA008_A_OPERATOR), 1, "RWA008: bad urn.can(operator)");
-        assertEq(rwaurn_008.can(RWA008_A_MATE),     0, "RWA008: bad urn.can(mate)");
-
-        assertEq(rwaconduitin_008.may(RWA008_A_OPERATOR), 1, "RWA008: bad inputConduit.may(operator)");
-        assertEq(rwaconduitin_008.may(RWA008_A_MATE),     1, "RWA008: bad inputConduit.may(mate)");
-
-        // We are not hope-ing the operator wallet in this spell because SocGen could not verify their addess in time.
-        assertEq(rwaconduitout_008.can(RWA008_A_OPERATOR), 0, "RWA008: bad outputConduit.can(operator)");
-        assertEq(rwaconduitout_008.can(RWA008_A_MATE),     0, "RWA008: bad outputConduit.can(mate)");
-        assertEq(rwaconduitout_008.may(RWA008_A_OPERATOR), 1, "RWA008: bad outputConduit.may(operator)");
-        assertEq(rwaconduitout_008.may(RWA008_A_MATE),     1, "RWA008: bad outputConduit.may(mate)");
-    }
-
-    function testRWA008_INTEGRATION_BUMP() public {
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
-
-        giveAuth(address(oracle), address(this));
-
-        (, address pip, , ) = oracle.ilks("RWA008-A");
-
-        assertEq(
-            DSValueAbstract(pip).read(),
-            bytes32(uint256(30_043_520_665599336150000000)),
-            "RWA008: Bad initial PIP value"
-        );
-
-        oracle.bump("RWA008-A", 40 * MILLION * WAD);
-
-        assertEq(DSValueAbstract(pip).read(), bytes32(40 * MILLION * WAD), "RWA008: Bad PIP value after bump()");
-    }
-
-    function testRWA008_INTEGRATION_TELL() public {
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
-
-        giveAuth(address(vat), address(this));
-        giveAuth(address(oracle), address(this));
-
-        (, , , uint48 tocPre) = oracle.ilks("RWA008-A");
-        assertEq(uint256(tocPre), 0, "RWA008: `toc` is not 0 before tell()");
-        assertTrue(oracle.good("RWA008-A"), "RWA008: Oracle not good before tell()");
-
-        vat.file("RWA008-A", "line", 0);
-        oracle.tell("RWA008-A");
-
-        (, , , uint48 tocPost) = oracle.ilks("RWA008-A");
-        assertGt(uint256(tocPost), 0, "RWA008: `toc` is not set after tell()");
-        assertTrue(!oracle.good("RWA008-A"), "RWA008: Oracle still good after tell()");
-    }
-
-    function testRWA008_INTEGRATION_TELL_CURE_GOOD() public {
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
-
-        giveAuth(address(vat), address(this));
-        giveAuth(address(oracle), address(this));
+    function test_RWA009Draw() public {
 
-        vat.file("RWA008-A", "line", 0);
-        oracle.tell("RWA008-A");
+        address dai         = addr.addr("MCD_DAI");
```

Comment:
`dai` is already declared as `DaiAbstract` in the base test. So you could avoid this line and not do the overwrite and the remove the casting in the following calls.

Reply:
done